plugins/base/src/main/kotlin/translators/descriptors/DescriptorAccessorConventionUtil.kt

```diff
@@ -0,0 +1,79 @@
+package org.jetbrains.dokka.base.translators.descriptors
+
+import org.jetbrains.dokka.base.translators.firstNotNullOfOrNull
+import org.jetbrains.kotlin.descriptors.FunctionDescriptor
+import org.jetbrains.kotlin.descriptors.PropertyDescriptor
+import org.jetbrains.kotlin.load.java.JvmAbi
+import org.jetbrains.kotlin.load.java.descriptors.JavaMethodDescriptor
+import org.jetbrains.kotlin.load.java.propertyNameByGetMethodName
+import org.jetbrains.kotlin.load.java.propertyNamesBySetMethodName
+
+internal data class DescriptorFunctionsHolder(
+    val regularFunctions: List<FunctionDescriptor>,
+    val accessors: Map<PropertyDescriptor, List<FunctionDescriptor>>
+)
+
+internal fun splitFunctionsAndAccessors(
```

Comment:
Yeah, unfortunately it looks like copy-paste, but in reality it's not really - it just uses the same principles

One deals with PSIs and the other one deals with Descriptors - I didn't find any common supertypes so that I could generify the logic. Arguably, I could add a bunch of our own abstractions and try to add complex generics, but it seems overkill for such a simple and small function

If you can suggest something more specific, we can try, but I couldn't find an elegant solution

Reply:
Unfortunately, current design depends on a low level things instead of our documenable model(


Maybe a inline function with reified types could be fit here.